tests/engine/recipes/test_default_recipe.py

```diff
@@ -161,6 +161,53 @@ def test_generate_graphs(
     rasa.engine.validation.validate(model_config)
 
 
+@pytest.mark.parametrize(
+    "cli_parameters, check_node, expected_config",
+    [
+        (
+            {},
+            "train_MitieIntentClassifier6",
+            {"num_threads": 200000, "finetuning_epoch_fraction": 0.75},
+        ),
+        (
+            {"num_threads": None},
+            "train_MitieIntentClassifier6",
+            {"num_threads": 200000, "finetuning_epoch_fraction": 0.75},
+        ),
+        (
+            {"num_threads": 1},
+            "train_MitieIntentClassifier6",
+            {"num_threads": 1, "finetuning_epoch_fraction": 0.75},
+        ),
+        (
+            {"num_threads": 1, "finetuning_epoch_fraction": 0.5},
+            "train_MitieIntentClassifier6",
+            # there is no `epochs` value specified so it doesn't get overridden
+            {"num_threads": 1, "finetuning_epoch_fraction": 0.75},
+        ),
+        (
+            {"finetuning_epoch_fraction": 0.5},
+            "train_DIETClassifier7",
+            {"epochs": 150, "num_threads": 200000, "finetuning_epoch_fraction": 0.5},
```

Comment:
What's the difference between DIET and Mitie classifiers? I mean both don't have `epochs` value specified in parameters so why Mitie has not been overwritten while DIET has been?

Reply:
@tayfun, it's because `MitieIntentClassifier` doesn't need `epochs` (more specifically, its default config doesn't contain `epochs`)